pkgs/pkgs-lib/formats.nix

```diff
@@ -142,6 +142,63 @@ rec {
 
   };
 
+  iniWithGlobalSection = {
```

Comment:
This function is looking good to me! However a lot of code is duplicated between it and the `ini` format. Can you de-duplicate that?

Reply:
I did deduplicate it, however I used an inline inherit statement with an attrset which *works* but isn't the most straightforward syntax, however I didn't want to introduce a global let statement at the top since it makes refactoring harder if the formats ever move to a separate file or such, and it has the potential for name collisions.

If the syntax is fine in your opinion please go ahead and resolve this, otherwise gimme a heads-up and I'll change it to a more conventional top-of-file let statement (or just move it to its own file like the Java properties).

By the way, this also moves the gitIni up into this so we can share the ini atoms.